app/controllers/tasks_controller.rb

```diff
@@ -1,8 +1,73 @@
 class TasksController < ApplicationController
+
   def index
+    @filter = params[:category]
+    @all_categories = %w[Inventory Register Engineering]
+    @tasks = if @filter.blank?
+               Task.all
+             else
+               Task.where(category: @filter)
+             end
+    incomplete_task = @tasks.where(completed: false).sort_by { |task| [-task.priority, task.added] }
```

Comment:
it's MUCH better to use `.order` to let the database do the sorting rather than sorting in ruby